src/components/Accordion/Accordion.tsx

```diff
@@ -224,16 +224,40 @@ export const Accordion: ChakraComponent<
       isDefaultOpen ? [0] : []
     );
 
+    const buttonRefs = useRef(accordionData.map(() => React.createRef()));
+
+    const updatedAccordionData = accordionData.map((content, i) => {
+      if (content.buttonInteractionRef) return content;
+
+      return { ...content, buttonInteractionRef: buttonRefs[i] };
+    });
+
     const handleKeyDown = (e) => {
-      // If the 'esc' key is pressed,
-      if (e.keyCode === 27) {
-        const focusedPanelIndex = Number(e.target.dataset.index);
-        // collapse the currently focused panel.
-        // (Nothing will happen if the currently
-        // focused panel is already collapsed.)
+      // If the 'esc' key is pressed, find the panel the
+      // user is focused on or within, and remove it as
+      // an expanded panel. (Nothing will happen if the
+      // panel is already collapsed.)
+      if (e.code === "Escape") {
```

Comment:
Just curious, why the update from `e.keyCode` to `e.code`?

Reply:
Two reasons. 1) I thought it was clearer than. using a keyCode, and 2) for some reason in the tests, the keyCode for Escape was a different number than it is in the code? So I used the code 'Escape' instead for the tests and wanted to be consistent.